Source/WebKit/NetworkProcess/storage/SQLiteStorageArea.cpp

```diff
@@ -405,5 +442,18 @@ void SQLiteStorageArea::handleLowMemoryWarning()
         m_database->releaseMemory();
 }
 
+void SQLiteStorageArea::handleDatabaseCorruption()
+{
+    ASSERT(!isMainRunLoop());
+    if (m_database->isOpen())
+        m_database->close();
+    m_database = nullptr;
+    RELEASE_LOG(Storage, "SQLiteStorageArea::handleDatabaseCorruption delete database file '%s'", m_path.utf8().data());
+    WebCore::SQLiteFileSystem::deleteDatabaseFile(m_path);
+
+    // Try to recreate the database.
+    prepareDatabase(ShouldCreateIfNotExists::Yes);
```

Comment:
I think we don't need `prepareDatabase` here because:
1. we are not retrying the failed operation on the new database; we will just return error, 
2. if client does not do any operation after the failed operation (that has led to database deletion), we don't want to have database open; if client performs new operation, `prepareDatabase` will be invoked on that operation.